src/uu/join/src/join.rs

```diff
@@ -775,8 +775,11 @@ fn get_field_number(keys: Option<usize>, key: Option<usize>) -> UResult<usize> {
 /// Parse the specified field string as a natural number and return
 /// the zero-based field number.
 fn parse_field_number(value: &str) -> UResult<usize> {
+    // TODO: use ParseIntError.kind() once MSRV >= 1.55
+    let overflow = "184467440737095516150".parse::<usize>().err().unwrap();
```

Comment:
This hack could potentially add some overhead, since we're probably doing this every time a field number gets parsed (I don't know how expensive these operations are, and I'm not sure how aggressive the compiler's optimizations are). In the typical case, this should be negligible, since this only happens a linear number of times in the size of the arguments, but someone could possibly be relying on some crazy xargs that causes it to show up. Possible workarounds would be passing `overflow` as an arg, storing it in global scope, or maybe a const fn (I'd have to check when things were added vs. MSRV), but my inclination is to keep the hack scoped to these lines, and just accept whatever overhead until either someone complains, or MSRV gets bumped enough we can use the [proper solution](https://doc.rust-lang.org/std/num/struct.ParseIntError.html#method.kind).

Reply:
That's a clever trick, hahaha. I can't believe we cannot properly introspect the kind until Rust 1.55. Even though it's not a `const fn`, it might get optimized out anyway. I think we can accept this if you add a bit of documentation on how this workaround works.

Reply:
I've added a bit of explanation, and a link to [where I found the trick](https://github.com/rust-lang/rust/issues/22639). But yeah, I suspect the actual runtime cost is either tiny or non-existent, I just haven't done any real investigation or benchmarking to confirm that.